### PR TITLE
RHOAIENG-4731 - Test gitops-update-pipeline as a part of PR check

### PR DESCRIPTION
**Pull request by MarianMacik**
**JIRA:** [RHOAIENG-4731](https://issues.redhat.com/browse/RHOAIENG-4731)
## Description
PR set:
* https://github.com/openshift/release/pull/50714
* https://github.com/opendatahub-io/ai-edge/pull/243

This PR adds a PR check to check the gitops-update-pipeline. It runs the pipeline, waits for the successful completion and finally checks the content of a created pull request.

2 things will likely change:
* ~Repository to use for the PR checks (currently my own fork)~ Now it's [redhat-rhods-qe/ai-edge-ci-test](https://github.com/redhat-rhods-qe/ai-edge-ci-test/pulls?q=is%3Apr+is%3Aclosed)
* ~Account to use for the PR creation (currently my account and my personal token)~ Now it's [ods-qe-test](https://github.com/ods-qe-test)

## How Has This Been Tested?
Locally and will be tested by the PR check.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: test/shell-pipeline-tests/README.md
Comment (LaVLaS) [inline test/shell-pipeline-tests/README.md:24]: Is there a technical restriction for using GitHub and a personal access token if we want to verify this locally?  I assume use of Gitea or gitlab for verification would also work assuming they are completely compatible with the GitOps pipeline codebase

Comment (MarianMacik) [reply test/shell-pipeline-tests/README.md:24]: @LaVLaS Not sure what you mean by `verify this locally`. We need a repository, which is reachable by the pipeline running on OpenShift, ideally one which can be accessed online using a browser for possible debugging etc.  Afaik our GitLab is behind VPN, isn't it? So it might not be reachable. I was hoping to use a separate GitHub repository with some kind of service account in the opendatahub-io org, there must be some already set up I expect.

Comment (LaVLaS) [reply test/shell-pipeline-tests/README.md:24]: Ok. The task name `github-open-pr` does imply that the workflow is github only but it does work with GitHub and Gitea (I have not tried GitLab) using Personal Access Tokens.  The original comment was in reference to use of `GitHub` but given the task name, I can understand why it was used

File: Makefile
Comment (grdryn) [inline Makefile:91]: Is the commenting out of this line expected to be committed, or was it just a local change while you were working on this?

File: test/shell-pipeline-tests/README.md
Comment (grdryn) [reply test/shell-pipeline-tests/README.md:24]: The task name having github specified in it is because it expects the GitHub API, which Gitea also happens to implement. I'm not sure if Gitlab or Bitbucket or other Git forges have the same API. If they don't, then a future improvement to the GitOps Update pipeline might be to add support for them.

I guess this credential description, and the token description on the following line, don't need to say _GitHub_ explicitly.

File: test/shell-pipeline-tests/common.sh
Comment (grdryn) [inline test/shell-pipeline-tests/common.sh:130]: There's a TODO here. Is that expected to be removed now, or at some future time? If it's in the future, is it something that could/should be tracked in a Jira issue that could be linked here?

File: Makefile
Comment (MarianMacik) [reply Makefile:91]: This is a part of the https://github.com/opendatahub-io/ai-edge/pull/243/commits/c6a399912fa21e7fe833032428c8063ecc9eedf2 commit which is here just for testing. It will be removed once PR check works.

File: test/shell-pipeline-tests/common.sh
Comment (MarianMacik) [reply test/shell-pipeline-tests/common.sh:130]: Yes, it is a reminder that this should be removed after #236 is merged as it removes this field.

File: test/shell-pipeline-tests/README.md
Comment (MarianMacik) [reply test/shell-pipeline-tests/README.md]: OK, I changed it to:
```
* **CUSTOM_GIT_CREDENTIALS_SECRET_PATH** - Directory where the GitHub-compatible forge token is stored. The directory should have 1 file:
  * token - containing the GitHub-compatible forge token, e.g. for GitHub specifically it will have the form of `github_pat_123...`
```
I didn't want to say just Git token as GitLab token won't work for opening the PR and the pipeline will fail.

**Comment by LaVLaS** [comment]
@MarianMacik Do you need to wait until openshift-ci issues are resolved before we can test & merge this PR?

File: test/shell-pipeline-tests/common.sh
Comment (grdryn) [inline test/shell-pipeline-tests/common.sh:167]: Notice the marker that GitHub (and even Git) put here at the end of the file when it's missing a newline at the end.

It's good practice to always have a newline at the end of files if possible, and you can typically configure your editor/IDE to ensure it's there. See this SO answer for more details: https://stackoverflow.com/a/729795

**Comment by MarianMacik** [comment]
@grdryn @LaVLaS feel free to review. All comments should be resolved now.

**Review by grdryn — APPROVED** [review]
/lgtm

@jackdelahunt FYI this might be a good one for you to review, given that you've been working on something similar in #255 for the Go tests

**Comment by LaVLaS** [comment]
/approve

Looks good. Great job @MarianMacik

**Pull request merged**
